src/chunk.rs

```diff
@@ -0,0 +1,73 @@
+use ::image::{ImageBuffer, Rgba, RgbaImage};
+use netcanv_renderer::paws::Point;
+use netcanv_renderer::{Framebuffer as FramebufferTrait, RenderBackend};
+
+use crate::backend::{Backend, Framebuffer};
+
+#[derive(Clone)]
+pub struct ChunkImage {
+   pub png: Vec<u8>,
+   pub webp: Option<Vec<u8>>,
+}
+
+/// A chunk on the infinite canvas.
+pub struct Chunk {
+   pub framebuffer: Framebuffer,
+   pub image_cache: Option<ChunkImage>,
+   saved: bool,
```

Comment:
Alright, so:

- We have the paint canvas, which handles actually painting the pixels to a framebuffer on the GPU, _maybe_ displaying them on the screen. (I'd prefer that to be its own thing but eh whatever, that isn't the point of this PR.)
- Then we'd have a layer on top of that: the image coder. The image coder wraps a paint canvas and gives it image encoding/decoding functionality.
- Then we'd have another layer on top of that: the image cache. It caches encoded images in a map separate from actual chunks.
- And then the newly named `ProjectFile` would be the one that keeps track of which chunks are saved and which are not.

Each layer introduces one bit of functionality on top of the existing chunks. It's a more compositional approach than before. Honestly I'm not sure if there's space for it in this PR since it'll be another large refactoring, so for now we can keep it as is; I was just throwing ideas at the wall here.